### PR TITLE
[PLATFORM-1443] Refactor stream listing table

### PR DESCRIPTION
**Pull request by juhah**
Changed stream list table to use CSS grid, now only one listing is needed to adapt to mobile, tablet & desktop views.

### CODE REVIEW

**Comment by juhah** [comment]
> Only thing I noticed was that the height of the mobile rows was inconsistent, depending on whether or not there was 'last data' content. This might have been always the case, hard to tell atm.

Yep, seems it was constant height before for all rows... will fix, unless @mattatgit this was the original design?

**Comment by mattatgit** [comment]
@juhah No, they should all be consistent heights.
Original design is here
https://share.goabstract.com/fba500db-d89e-4d78-a1ad-9fcf029db1f4?mode=design&sha=0858b2bb5956bbc82cfceb923f5d1b205e363fc5

Couple of notes to that design
* i think the status lights should just be vertically centred in the list element
* i don't think we should be using the signed blue tick because soon we'll be requiring all streams to be signed so it will be a bit redundant.

**Comment by juhah** [comment]
> i think the status lights should just be vertically centred in the list element
> i don't think we should be using the signed blue tick because soon we'll be requiring all streams to be signed so it will be a bit redundant.

@mattatgit Ok, I fixed up the height & icon: 
Screenshot 2020-07-10 at 8 55 13

Tablet:
Screenshot 2020-07-10 at 8 55 26

Desktop:
Screenshot 2020-07-10 at 8 55 49

**Comment by mattatgit** [comment]
@juhah Tbh, there's a lot of things need fixing there.

1. Mobile needs more aggressive truncation, small screens, lines need to be truncated at shorter length
2. Line spacing / vertical padding between name & description don't match the design, much wider than design
3. We should remove the green padlock / encrypted icon for now, its not being used in other areas, so please remove & we'll re-implement consistently across the platform

I think actually the design needs revisiting & some changes, i just see too many issues here. 
Can we pause this for a few days while we go in & fix a few things ?

**Comment by juhah** [comment]
> Can we pause this for a few days while we go in & fix a few things ?

@mattatgit Sure, we can leave this for now - I'll be anyway afk for a couple of week and can revisit this after 👍

**Comment by mattatgit** [comment]
@juhah Ok, no worries. We'll sort it out next week & have it ready for you when you get back. Have a good one!

**Comment by fonty1** [comment]
I can probably take over if we want to jump on this sooner, depending on how long we pause this for ofc. Let's play it by ear 👍

**Comment by mattatgit** [comment]
@fonty1 Yep, no worries. Won't take long, will be ready on Monday, so if you want to pick it up then or Tuesday, would be fine.

File: app/src/userpages/components/StreamPage/List/index.jsx [hunk 1]
```diff
@@ -51,26 +47,341 @@ import Sidebar from '$shared/components/Sidebar'
...
+
+        .dropdown {
+            visibility: visible;
+        }
+    }
+`
+
+const RowItem = styled.div`
+    display: none;
+    overflow-y: visible;
```
Comment (mondoreale) [inline app/src/userpages/components/StreamPage/List/index.jsx:168]: Selective `overflow` means potential trouble in the future. What problem does it solve?

File: app/src/userpages/components/StreamPage/List/index.jsx [hunk 2]
```diff
@@ -51,26 +47,341 @@ import Sidebar from '$shared/components/Sidebar'
...
+    color: #a3a3a3;
+
+    @media (min-width: ${MD}px) {
+        display: block;
+        overflow-x: hidden;
+        word-break: break-word;
+        text-overflow: ellipsis;
+        white-space: nowrap;
+        grid-row: 2;
+        grid-column: 1;
```
Comment (mondoreale) [inline app/src/userpages/components/StreamPage/List/index.jsx:218]: We're getting quite specific here. I'll have to check it out in the browser. 👀

File: app/src/userpages/components/StreamPage/List/index.jsx [hunk 3]
```diff
@@ -51,26 +47,341 @@ import Sidebar from '$shared/components/Sidebar'
...
+    word-break: break-word;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+`
+
+const SecurityIconContainer = styled.span`
+    position: relative;
+    line-height: 12px;
+
+    &::after {
```
Comment (mondoreale) [inline app/src/userpages/components/StreamPage/List/index.jsx:347]: Another custom tooltip. Do we have to follow that path?

File: app/src/userpages/components/StreamPage/List/index.jsx [hunk 4]
```diff
@@ -51,26 +47,341 @@ import Sidebar from '$shared/components/Sidebar'
...
+    }
+
+    @media (min-width: ${LG}px) {
+        && {
+            margin-bottom: 0;
+        }
+    }
+`
+
+const LoadingMoreSpacer = styled.div`
```
Comment (mondoreale) [inline app/src/userpages/components/StreamPage/List/index.jsx:112]: I now see it's a space "holder". How about we bake that into `LoadMore` via a prop like `preserveSpace` (or similar)? Current way makes us having to customize it for every `LoadMore` instance if we need it.

File: app/src/shared/components/StatusIcon/statusIcon.pcss [hunk 5]
```diff
@@ -22,6 +22,7 @@
     left: 50%;
     transform: translateX(-50%);
     padding: 2px 6px;
+    white-space: nowrap;
```
[the remark this answers is not in the document]
Comment (juhah) [reply app/src/shared/components/StatusIcon/statusIcon.pcss:25]: I agree, we should just use one :D This stream list variant is even different between the security icon and the status icon (see staging or production).

File: app/src/userpages/components/StreamPage/List/index.jsx [hunk 6]
```diff
@@ -51,26 +47,341 @@ import Sidebar from '$shared/components/Sidebar'
...
+`
+
+const StyledListContainer = styled(ListContainer)`
+    && {
+        padding: 0;
+        margin-bottom: 4em;
+    }
+
+    ${StreamTable}:last-child {
+        margin-bottom: 4rem;
```
[the remark this answers is not in the document]
Comment (juhah) [reply app/src/userpages/components/StreamPage/List/index.jsx:95]: Yep, counteracting some styles that I was afraid to remove so they don't break things elsewhere. Should go through them again and see which places are affected.

File: app/src/userpages/components/StreamPage/List/index.jsx (hunk 1 above)
Comment (juhah) [reply app/src/userpages/components/StreamPage/List/index.jsx:168]: I think this is no longer needed

File: app/src/userpages/components/StreamPage/List/index.jsx (hunk 4 above)
Comment (juhah) [reply app/src/userpages/components/StreamPage/List/index.jsx:112]: agreed

**Comment by juhah** [comment]
@fonty1 Go ahead if you want to pick this up 👍 Otherwise I will continue on it later

**Comment by mondoreale** [comment]
@juhah, all good sir!

Have a decent break!

We're gonna take care of this puppy while you're gone. 🐶

Cheers!

**Comment by Saorikaji2** [comment]
@fonty1 
Here are the latest Core Streams responsive views.
Notable changes
Tablet:
* “Last data” will be next to the stream name
* Size of status light will be 16x16
Mobile:
* Size of status light will be 16x16

https://www.figma.com/file/rYYvVreKa7iiABu6Nl9OUf/Core-app-2020?node-id=150%3A314

**Comment by fonty1** [comment]
> @fonty1
> Here are the latest Core Streams responsive views.
> Notable changes
> Tablet:
> 
> * “Last data” will be next to the stream name
> * Size of status light will be 16x16
>   Mobile:
> * Size of status light will be 16x16
> 
> https://www.figma.com/file/rYYvVreKa7iiABu6Nl9OUf/Core-app-Q3-2020?node-id=150%3A314

Thanks @Saorikaji2 looks good.

**Comment by juhah** [comment]
@fonty1 @mondoreale I updated the list with the newer design, please have another look when you have the time.

**Comment by fonty1** [comment]
LGTM @juhah - I think the design intention was to remove the green lock icon too though, so I think that needs to go.

**Comment by juhah** [comment]
> I think the design intention was to remove the green lock icon too though, so I think that needs to go.

Ok, it has been told to go and has left the premises.

**Comment by mondoreale** [comment]
Haha, I literally was looking at this PR when you merged it. Glad u did! 🚀

**Comment by mattatgit** [comment]
@mondoreale yep if no description, let's centre them.

**Comment by juhah** [comment]
@mattatgit @mondoreale ok, will address the centering in another issue!

**Pull request merged**